csrc/fusion_segmenter.cpp

```diff
@@ -4356,12 +4374,28 @@ void SegmentCandidateFinder::forwardInputs() {
   excluded_inp_unary_exprs_ = {};
   input2group_.clear();
 
+  std::vector<Val*> extended_fusion_inputs = completeFusion()->inputs();
+
+  // Grab factory ops that should be forwarded. Add created tensors to
+  // the fusion input list to make them handled like fusion inputs
+  // TODO: Handle more factory methods such as IotaOp, EyeOp,
+  // TensorConstruct. Probably should not include relatively expensive
+  // ops like RNGOp.
+  for (auto expr : completeFusion()->exprs()) {
+    if (expr->isA<FullOp>() &&
+        // Don't bother if it's a fusion output
+        !expr->output(0)->isFusionOutput()) {
```

Comment:
should we not forward even though they are fusion outputs, that would still save bandwidth on reading gmem in other consumers right?!

Reply:
That's true in theory, but in practice it'll be likely cached anyway, so the impact would be minimal. 

Implementation wise, we would need to further extend the forwarding logic since it's likely to result in causing something unexpected if multiple segments had the same op producing the same fusion output.

I just don't think it's worthwhile supporting such a case.